### PR TITLE
Refactor module for Puppet 6 & 7

### PR DESCRIPTION
**Pull request by Phil-Friderici**
Will collect the commits here.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -354,9 +353,9 @@
         # to ensure the start script is found on the non-standard locations.
         if $::operatingsystem != 'Ubuntu' {
           Service[$service_name_real] {
-            start  =>  "${service_path}${_service_name_default} start",
-            stop   =>  "${service_path}${_service_name_default} stop",
-            status =>  "${service_path}${_service_name_default} status",
+            start  =>  "${service_path}vmware-tools-services start",
```
Comment (anders-larsson) [inline manifests/init.pp:240]: Remove final / from $service_path if possible.

File: manifests/init.pp [hunk 2]
```diff
@@ -127,7 +127,7 @@
   Boolean              $manage_tools_nox_package      = true,
   Boolean              $disable_tools_version         = true,
   String[1]            $esx_version                   = 'latest',
-  String[1]            $proxy_host                    = 'absent',
+  Optional[Boolean]    $proxy_host                    = undef,
```
Comment (anders-larsson) [inline manifests/init.pp:152]: proxy_host must be a hostname

File: REFERENCE.md [hunk 3]
```diff
@@ -200,14 +223,16 @@ Default value: ``undef``
 Data type: `String[1]`
 
 String to pass to ensure attribute for the vmwaretools x package.
+Use 'present', 'latest' to install or 'absent', 'purged' to remove package.
```
Comment (anders-larsson) [inline REFERENCE.md:226]: Enum['...']

**Review by anders-larsson — CHANGES_REQUESTED** [review]
Discussed changes with Phil.

File: README.md [hunk 4]
```diff
@@ -1,154 +1,34 @@
...
+
+This module has been tested to work on the following systems with Puppet
+versions 6 and 7 with the Ruby version associated with those releases.
+This module aims to support the current and previous major Puppet versions.
+
+ * RedHat/CentOS 5
+ * RedHat/CentOS 6
+ * RedHat/CentOS 7
+ * RedHat/CentOS 8
+ # Suse (SLED/SLES) 10
```
[the remark this answers is not in the document]
Comment (Phil-Friderici) [reply README.md]: Changed the # to a *.
Unsure myself about pointing to metadata here.
As technical person I am ok to have a quick look in metadata.json.
Would spare some changes in README as plus and README wouldn't become outdated for the supported OSes anymore too.

File: manifests/init.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/init.pp:240]: done with a `regsubst()`

File: REFERENCE.md (hunk 3 above)
Comment (Phil-Friderici) [reply REFERENCE.md:226]: In the old tests is test that uses explicit version numbers:


```
  context 'with custom values for parameters on machine running on vmware' do
    let(:facts) { [default_facts, { vmware_has_x: true }].reduce(:merge) }
    let(:params) do
      {
        tools_nox_package_name: 'vmware-tools-esx-nox-custom',
        tools_nox_package_ensure: '0.2-1',
        tools_x_package_name: 'vmware-tools-esx-custom',
        tools_x_package_ensure: '0.3-1',
      }
    end

    it { is_expected.to contain_package('vmware-tools-esx-nox-custom').with('ensure' => '0.2-1') }
    it { is_expected.to contain_package('vmware-tools-esx-custom').with('ensure' => '0.3-1') }
  end
```

I could add a regex that will match `[numbers].[numbers]-[numbers]` or `[numbers].[numbers].*` to also allow them.

File: manifests/init.pp (hunk 2 above)
Comment (Phil-Friderici) [reply manifests/init.pp]: fixed, now it's `Stdlib::Host`

File: REFERENCE.md (hunk 3 above)
Comment (Phil-Friderici) [reply REFERENCE.md]: Fixed with `Variant[Enum['absent', 'latest', 'present', 'purged'], Pattern[/(\d+\.)+([\d-]+)/]]`.
The Regex allows strings with version numbers in the given format (eg. nn.nn-n, nn.nn.nn nn.nn nn.nn.nn-n)

File: manifests/repo/debian.pp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  String[1]            $esx_version                   = 'latest',
+  Optional[Boolean]    $proxy_host                    = undef,
+  Integer[0, 65535]    $proxy_port                    = 8080,
+){
+
+  if $proxy_host == undef {
+    include ::apt
+  } else {
+    # will only work if apt is not already defined elsewhere
+    class { '::apt':
```
[the remark this answers is not in the document]
Comment (Phil-Friderici) [reply manifests/repo/debian.pp]: As discussed with @anders-larsson the proxy functionality is removed.
Proxy should instead should be set system wide or via package manager.

**Comment by Phil-Friderici** [comment]
@anders-larsson as discussed earlier, the proxy_* functionality for package managers got removed completely from the module. I also finished to update the dependency modules.

Would recommend to test these changes one more time in production. When everything works as expected, refactor tests and release v2.0.0 \o/

**Comment by Phil-Friderici** [comment]
Merged into master branch after second review with @anders-larsson.

**Pull request merged**